### ISSUE TITLE
Issues on Startup (Fabric 1.21.1 and 0.16.9)

### ISSUE THREAD

**Issue by suerion** [Meek97/Better-X-BOP#1]
I think, you should look again on your Datapack, does not work for me on registry

Latest.log:
https://mclo.gs/F165452

```
> Errors in registry minecraft:root:
>> Errors in element minecraft:worldgen/configured_feature:
java.lang.IllegalStateException: Unbound values in registry ResourceKey[minecraft:root / minecraft:worldgen/configured_feature]: [betterxbop:patch_basalt_stalactite, betterxbop:patch_basalt_stalagmite, betterxbop:patch_blackstone_stalactite_on_ceil, betterxbop:patch_blackstone_stalagmite_on_ground]
        at knot//net.minecraft.class_2370.method_40276(class_2370.java:355)
        at knot//net.minecraft.class_7655.method_45128(class_7655.java:169)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at knot//net.minecraft.class_7655.method_45121(class_7655.java:166)
        at knot//net.minecraft.class_7655.mixinextras$bridge$method_45121$49(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall(class_7655.java:2056)
        at knot//net.minecraft.class_7655.mixinextras$bridge$wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall$52(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$dka000$fabric-resource-conditions-api-v1$captureRegistries(class_7655.java:2557)
        at knot//net.minecraft.class_7655.method_56515(class_7655.java:151)
        at knot//net.minecraft.class_7237.method_45142(class_7237.java:62)
        at knot//net.minecraft.class_7237.method_45143(class_7237.java:66)
        at knot//net.minecraft.class_7237.method_42098(class_7237.java:32)
        at knot//net.minecraft.server.Main.method_43612(Main.java:184)
        at knot//net.minecraft.class_156.method_43498(class_156.java:1018)
        at knot//net.minecraft.class_156.method_43499(class_156.java:1012)
        at knot//net.minecraft.server.Main.main(Main.java:182)
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23)
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69)
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69)
>> Errors in element minecraft:worldgen/placed_feature:
java.lang.IllegalStateException: Unbound values in registry ResourceKey[minecraft:root / minecraft:worldgen/placed_feature]: [betterxbop:patch_basalt_stalactite_sparse, betterxbop:patch_blackstone_stalactite_on_ceil]
        at knot//net.minecraft.class_2370.method_40276(class_2370.java:355)
        at knot//net.minecraft.class_7655.method_45128(class_7655.java:169)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at knot//net.minecraft.class_7655.method_45121(class_7655.java:166)
        at knot//net.minecraft.class_7655.mixinextras$bridge$method_45121$49(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall(class_7655.java:2056)
        at knot//net.minecraft.class_7655.mixinextras$bridge$wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall$52(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$dka000$fabric-resource-conditions-api-v1$captureRegistries(class_7655.java:2557)
        at knot//net.minecraft.class_7655.method_56515(class_7655.java:151)
        at knot//net.minecraft.class_7237.method_45142(class_7237.java:62)
        at knot//net.minecraft.class_7237.method_45143(class_7237.java:66)
        at knot//net.minecraft.class_7237.method_42098(class_7237.java:32)
        at knot//net.minecraft.server.Main.method_43612(Main.java:184)
        at knot//net.minecraft.class_156.method_43498(class_156.java:1018)
        at knot//net.minecraft.class_156.method_43499(class_156.java:1012)
        at knot//net.minecraft.server.Main.main(Main.java:182)
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23)
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69)
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69)
> Errors in registry minecraft:worldgen/configured_feature:
>> Errors in element betterxbop:patch_basalt_stalactite:
java.lang.IllegalStateException: Failed to parse betterxbop:worldgen/configured_feature/patch_basalt_stalactite.json from pack file/Better-X-BOP_1.1_1.21.1.zip
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:251)
        at knot//net.minecraft.class_7655$class_9158.method_56520(class_7655.java:96)
        at knot//net.minecraft.class_7655.method_56514(class_7655.java:151)
        at knot//net.minecraft.class_7655.method_45120(class_7655.java:164)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at knot//net.minecraft.class_7655.method_45121(class_7655.java:164)
        at knot//net.minecraft.class_7655.mixinextras$bridge$method_45121$49(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall(class_7655.java:2056)
        at knot//net.minecraft.class_7655.mixinextras$bridge$wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall$52(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$dka000$fabric-resource-conditions-api-v1$captureRegistries(class_7655.java:2557)
        at knot//net.minecraft.class_7655.method_56515(class_7655.java:151)
        at knot//net.minecraft.class_7237.method_45142(class_7237.java:62)
        at knot//net.minecraft.class_7237.method_45143(class_7237.java:66)
        at knot//net.minecraft.class_7237.method_42098(class_7237.java:32)
        at knot//net.minecraft.server.Main.method_43612(Main.java:184)
        at knot//net.minecraft.class_156.method_43498(class_156.java:1018)
        at knot//net.minecraft.class_156.method_43499(class_156.java:1012)
        at knot//net.minecraft.server.Main.main(Main.java:182)
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23)
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69)
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69)
Caused by: java.lang.IllegalStateException: Unknown registry key in ResourceKey[minecraft:root / minecraft:worldgen/placement_modifier_type]: bclib:is; Unknown registry key in ResourceKey[minecraft:root / minecraft:worldgen/feature]: bclib:pillar
        at knot//com.mojang.serialization.DataResult$Error.md65ef26$owo$lambda$addStackTraceToException$0$0(DataResult.java:565)
        at knot//com.mojang.serialization.DataResult$Error.getOrThrow(DataResult.java:287)
        at knot//com.mojang.serialization.DataResult.getOrThrow(DataResult.java:81)
        at knot//net.minecraft.class_7655.method_56915(class_7655.java:231)
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:249)
        ... 22 more
>> Errors in element betterxbop:patch_basalt_stalagmite:
java.lang.IllegalStateException: Failed to parse betterxbop:worldgen/configured_feature/patch_basalt_stalagmite.json from pack file/Better-X-BOP_1.1_1.21.1.zip
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:251)
        at knot//net.minecraft.class_7655$class_9158.method_56520(class_7655.java:96)
        at knot//net.minecraft.class_7655.method_56514(class_7655.java:151)
        at knot//net.minecraft.class_7655.method_45120(class_7655.java:164)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at knot//net.minecraft.class_7655.method_45121(class_7655.java:164)
        at knot//net.minecraft.class_7655.mixinextras$bridge$method_45121$49(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall(class_7655.java:2056)
        at knot//net.minecraft.class_7655.mixinextras$bridge$wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall$52(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$dka000$fabric-resource-conditions-api-v1$captureRegistries(class_7655.java:2557)
        at knot//net.minecraft.class_7655.method_56515(class_7655.java:151)
        at knot//net.minecraft.class_7237.method_45142(class_7237.java:62)
        at knot//net.minecraft.class_7237.method_45143(class_7237.java:66)
        at knot//net.minecraft.class_7237.method_42098(class_7237.java:32)
        at knot//net.minecraft.server.Main.method_43612(Main.java:184)
        at knot//net.minecraft.class_156.method_43498(class_156.java:1018)
        at knot//net.minecraft.class_156.method_43499(class_156.java:1012)
        at knot//net.minecraft.server.Main.main(Main.java:182)
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23)
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69)
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69)
Caused by: java.lang.IllegalStateException: Unknown registry key in ResourceKey[minecraft:root / minecraft:worldgen/placement_modifier_type]: bclib:is; Unknown registry key in ResourceKey[minecraft:root / minecraft:worldgen/feature]: bclib:pillar
        at knot//com.mojang.serialization.DataResult$Error.md65ef26$owo$lambda$addStackTraceToException$0$0(DataResult.java:565)
        at knot//com.mojang.serialization.DataResult$Error.getOrThrow(DataResult.java:287)
        at knot//com.mojang.serialization.DataResult.getOrThrow(DataResult.java:81)
        at knot//net.minecraft.class_7655.method_56915(class_7655.java:231)
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:249)
        ... 22 more
>> Errors in element betterxbop:patch_blackstone_stalactite_on_ceil:
java.lang.IllegalStateException: Failed to parse betterxbop:worldgen/configured_feature/patch_blackstone_stalactite_on_ceil.json from pack file/Better-X-BOP_1.1_1.21.1.zip
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:251)
        at knot//net.minecraft.class_7655$class_9158.method_56520(class_7655.java:96)
        at knot//net.minecraft.class_7655.method_56514(class_7655.java:151)
        at knot//net.minecraft.class_7655.method_45120(class_7655.java:164)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at knot//net.minecraft.class_7655.method_45121(class_7655.java:164)
        at knot//net.minecraft.class_7655.mixinextras$bridge$method_45121$49(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall(class_7655.java:2056)
        at knot//net.minecraft.class_7655.mixinextras$bridge$wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall$52(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$dka000$fabric-resource-conditions-api-v1$captureRegistries(class_7655.java:2557)
        at knot//net.minecraft.class_7655.method_56515(class_7655.java:151)
        at knot//net.minecraft.class_7237.method_45142(class_7237.java:62)
        at knot//net.minecraft.class_7237.method_45143(class_7237.java:66)
        at knot//net.minecraft.class_7237.method_42098(class_7237.java:32)
        at knot//net.minecraft.server.Main.method_43612(Main.java:184)
        at knot//net.minecraft.class_156.method_43498(class_156.java:1018)
        at knot//net.minecraft.class_156.method_43499(class_156.java:1012)
        at knot//net.minecraft.server.Main.main(Main.java:182)
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23)
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69)
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69)
Caused by: java.lang.IllegalStateException: Unknown registry key in ResourceKey[minecraft:root / minecraft:worldgen/placement_modifier_type]: bclib:is; Unknown registry key in ResourceKey[minecraft:root / minecraft:worldgen/feature]: bclib:pillar
        at knot//com.mojang.serialization.DataResult$Error.md65ef26$owo$lambda$addStackTraceToException$0$0(DataResult.java:565)
        at knot//com.mojang.serialization.DataResult$Error.getOrThrow(DataResult.java:287)
        at knot//com.mojang.serialization.DataResult.getOrThrow(DataResult.java:81)
        at knot//net.minecraft.class_7655.method_56915(class_7655.java:231)
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:249)
        ... 22 more
>> Errors in element betterxbop:patch_blackstone_stalagmite_on_ground:
java.lang.IllegalStateException: Failed to parse betterxbop:worldgen/configured_feature/patch_blackstone_stalagmite_on_ground.json from pack file/Better-X-BOP_1.1_1.21.1.zip
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:251)
        at knot//net.minecraft.class_7655$class_9158.method_56520(class_7655.java:96)
        at knot//net.minecraft.class_7655.method_56514(class_7655.java:151)
        at knot//net.minecraft.class_7655.method_45120(class_7655.java:164)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at knot//net.minecraft.class_7655.method_45121(class_7655.java:164)
        at knot//net.minecraft.class_7655.mixinextras$bridge$method_45121$49(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall(class_7655.java:2056)
        at knot//net.minecraft.class_7655.mixinextras$bridge$wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall$52(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$dka000$fabric-resource-conditions-api-v1$captureRegistries(class_7655.java:2557)
        at knot//net.minecraft.class_7655.method_56515(class_7655.java:151)
        at knot//net.minecraft.class_7237.method_45142(class_7237.java:62)
        at knot//net.minecraft.class_7237.method_45143(class_7237.java:66)
        at knot//net.minecraft.class_7237.method_42098(class_7237.java:32)
        at knot//net.minecraft.server.Main.method_43612(Main.java:184)
        at knot//net.minecraft.class_156.method_43498(class_156.java:1018)
        at knot//net.minecraft.class_156.method_43499(class_156.java:1012)
        at knot//net.minecraft.server.Main.main(Main.java:182)
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23)
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69)
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69)
Caused by: java.lang.IllegalStateException: Unknown registry key in ResourceKey[minecraft:root / minecraft:worldgen/feature]: bclib:sequence
        at knot//com.mojang.serialization.DataResult$Error.md65ef26$owo$lambda$addStackTraceToException$0$0(DataResult.java:565)
        at knot//com.mojang.serialization.DataResult$Error.getOrThrow(DataResult.java:287)
        at knot//com.mojang.serialization.DataResult.getOrThrow(DataResult.java:81)
        at knot//net.minecraft.class_7655.method_56915(class_7655.java:231)
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:249)
        ... 22 more
> Errors in registry minecraft:worldgen/placed_feature:
>> Errors in element betterxbop:patch_basalt_stalactite:
java.lang.IllegalStateException: Failed to parse betterxbop:worldgen/placed_feature/patch_basalt_stalactite.json from pack file/Better-X-BOP_1.1_1.21.1.zip
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:251)
        at knot//net.minecraft.class_7655$class_9158.method_56520(class_7655.java:96)
        at knot//net.minecraft.class_7655.method_56514(class_7655.java:151)
        at knot//net.minecraft.class_7655.method_45120(class_7655.java:164)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at knot//net.minecraft.class_7655.method_45121(class_7655.java:164)
        at knot//net.minecraft.class_7655.mixinextras$bridge$method_45121$49(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall(class_7655.java:2056)
        at knot//net.minecraft.class_7655.mixinextras$bridge$wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall$52(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$dka000$fabric-resource-conditions-api-v1$captureRegistries(class_7655.java:2557)
        at knot//net.minecraft.class_7655.method_56515(class_7655.java:151)
        at knot//net.minecraft.class_7237.method_45142(class_7237.java:62)
        at knot//net.minecraft.class_7237.method_45143(class_7237.java:66)
        at knot//net.minecraft.class_7237.method_42098(class_7237.java:32)
        at knot//net.minecraft.server.Main.method_43612(Main.java:184)
        at knot//net.minecraft.class_156.method_43498(class_156.java:1018)
        at knot//net.minecraft.class_156.method_43499(class_156.java:1012)
        at knot//net.minecraft.server.Main.main(Main.java:182)
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23)
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69)
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69)
Caused by: java.lang.IllegalStateException: Unknown registry key in ResourceKey[minecraft:root / minecraft:worldgen/placement_modifier_type]: bclib:under_every_layer
        at knot//com.mojang.serialization.DataResult$Error.md65ef26$owo$lambda$addStackTraceToException$0$0(DataResult.java:565)
        at knot//com.mojang.serialization.DataResult$Error.getOrThrow(DataResult.java:287)
        at knot//com.mojang.serialization.DataResult.getOrThrow(DataResult.java:81)
        at knot//net.minecraft.class_7655.method_56915(class_7655.java:231)
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:249)
        ... 22 more
>> Errors in element betterxbop:patch_basalt_stalactite_sparse:
java.lang.IllegalStateException: Failed to parse betterxbop:worldgen/placed_feature/patch_basalt_stalactite_sparse.json from pack file/Better-X-BOP_1.1_1.21.1.zip
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:251)
        at knot//net.minecraft.class_7655$class_9158.method_56520(class_7655.java:96)
        at knot//net.minecraft.class_7655.method_56514(class_7655.java:151)
        at knot//net.minecraft.class_7655.method_45120(class_7655.java:164)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at knot//net.minecraft.class_7655.method_45121(class_7655.java:164)
        at knot//net.minecraft.class_7655.mixinextras$bridge$method_45121$49(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall(class_7655.java:2056)
        at knot//net.minecraft.class_7655.mixinextras$bridge$wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall$52(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$dka000$fabric-resource-conditions-api-v1$captureRegistries(class_7655.java:2557)
        at knot//net.minecraft.class_7655.method_56515(class_7655.java:151)
        at knot//net.minecraft.class_7237.method_45142(class_7237.java:62)
        at knot//net.minecraft.class_7237.method_45143(class_7237.java:66)
        at knot//net.minecraft.class_7237.method_42098(class_7237.java:32)
        at knot//net.minecraft.server.Main.method_43612(Main.java:184)
        at knot//net.minecraft.class_156.method_43498(class_156.java:1018)
        at knot//net.minecraft.class_156.method_43499(class_156.java:1012)
        at knot//net.minecraft.server.Main.main(Main.java:182)
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23)
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69)
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69)
Caused by: java.lang.IllegalStateException: Unknown registry key in ResourceKey[minecraft:root / minecraft:worldgen/placement_modifier_type]: bclib:under_every_layer
        at knot//com.mojang.serialization.DataResult$Error.md65ef26$owo$lambda$addStackTraceToException$0$0(DataResult.java:565)
        at knot//com.mojang.serialization.DataResult$Error.getOrThrow(DataResult.java:287)
        at knot//com.mojang.serialization.DataResult.getOrThrow(DataResult.java:81)
        at knot//net.minecraft.class_7655.method_56915(class_7655.java:231)
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:249)
        ... 22 more
>> Errors in element betterxbop:patch_blackstone_stalactite_on_ceil:
java.lang.IllegalStateException: Failed to parse betterxbop:worldgen/placed_feature/patch_blackstone_stalactite_on_ceil.json from pack file/Better-X-BOP_1.1_1.21.1.zip
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:251)
        at knot//net.minecraft.class_7655$class_9158.method_56520(class_7655.java:96)
        at knot//net.minecraft.class_7655.method_56514(class_7655.java:151)
        at knot//net.minecraft.class_7655.method_45120(class_7655.java:164)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at knot//net.minecraft.class_7655.method_45121(class_7655.java:164)
        at knot//net.minecraft.class_7655.mixinextras$bridge$method_45121$49(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall(class_7655.java:2056)
        at knot//net.minecraft.class_7655.mixinextras$bridge$wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall$52(class_7655.java)
        at knot//net.minecraft.class_7655.wrapOperation$dka000$fabric-resource-conditions-api-v1$captureRegistries(class_7655.java:2557)
        at knot//net.minecraft.class_7655.method_56515(class_7655.java:151)
        at knot//net.minecraft.class_7237.method_45142(class_7237.java:62)
        at knot//net.minecraft.class_7237.method_45143(class_7237.java:66)
        at knot//net.minecraft.class_7237.method_42098(class_7237.java:32)
        at knot//net.minecraft.server.Main.method_43612(Main.java:184)
        at knot//net.minecraft.class_156.method_43498(class_156.java:1018)
        at knot//net.minecraft.class_156.method_43499(class_156.java:1012)
        at knot//net.minecraft.server.Main.main(Main.java:182)
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23)
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69)
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69)
Caused by: java.lang.IllegalStateException: Unknown registry key in ResourceKey[minecraft:root / minecraft:worldgen/placement_modifier_type]: bclib:under_every_layer
        at knot//com.mojang.serialization.DataResult$Error.md65ef26$owo$lambda$addStackTraceToException$0$0(DataResult.java:565)
        at knot//com.mojang.serialization.DataResult$Error.getOrThrow(DataResult.java:287)
        at knot//com.mojang.serialization.DataResult.getOrThrow(DataResult.java:81)
        at knot//net.minecraft.class_7655.method_56915(class_7655.java:231)
        at knot//net.minecraft.class_7655.method_45122(class_7655.java:249)
        ... 22 more

[20:51:41] [main/WARN]: Failed to load datapacks, can't proceed with server load. You can either fix your datapacks or reset to vanilla with --safeMode
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Failed to load registries due to above errors
        at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396) ~[?:?]
        at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073) ~[?:?]
        at knot/net.minecraft.server.Main.main(Main.java:229) [server-intermediary.jar:?]
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) [fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.16.9.jar:?]
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) [fabric-loader-0.16.9.jar:?]
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69) [fabric-server-mc.1.21.1-loader.0.16.9-launcher.1.0.1.jar:1.0.1]
Caused by: java.lang.IllegalStateException: Failed to load registries due to above errors
        at knot//MC/net.minecraft.registry.RegistryLoader.load(RegistryLoader.java:181) ~[server-intermediary.jar:?]
        at knot//MC/net.minecraft.registry.RegistryLoader.mixinextras$bridge$method_45121$49(RegistryLoader.java) ~[server-intermediary.jar:?]
        at knot//MC/net.minecraft.registry.RegistryLoader.wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall(RegistryLoader.java:2056) ~[server-intermediary.jar:?]
        at knot//MC/net.minecraft.registry.RegistryLoader.mixinextras$bridge$wrapOperation$djk000$fabric-registry-sync-v0$wrapIsServerCall$52(RegistryLoader.java) ~[server-intermediary.jar:?]
        at knot//MC/net.minecraft.registry.RegistryLoader.wrapOperation$dka000$fabric-resource-conditions-api-v1$captureRegistries(RegistryLoader.java:2557) ~[server-intermediary.jar:?]
        at knot//MC/net.minecraft.registry.RegistryLoader.loadFromResource(RegistryLoader.java:151) ~[server-intermediary.jar:?]
        at knot//MC/net.minecraft.server.SaveLoading.loadDynamicRegistryManager(SaveLoading.java:62) ~[server-intermediary.jar:?]
        at knot//MC/net.minecraft.server.SaveLoading.withRegistriesLoaded(SaveLoading.java:66) ~[server-intermediary.jar:?]
        at knot//MC/net.minecraft.server.SaveLoading.load(SaveLoading.java:32) ~[server-intermediary.jar:?]
        at knot/net.minecraft.server.Main.method_43612(Main.java:184) ~[server-intermediary.jar:?]
        at knot//MC/net.minecraft.util.Util.waitAndApply(Util.java:1018) ~[server-intermediary.jar:?]
        at knot//MC/net.minecraft.util.Util.waitAndApply(Util.java:1012) ~[server-intermediary.jar:?]
        at knot/net.minecraft.server.Main.main(Main.java:182) ~[server-intermediary.jar:?]
        ... 5 more
```

**Comment by suerion** [comment]
@Meek97 did you see this issue?

**Comment by Meek97** [comment]
Hey, sorry for the delay.
Yeah I can definitely see that there are problems with the datapack versions available from Modrinth. This is my first datapack upload, and I threw the 1.21.1 changes together pretty quickly after getting the 1.20.1 version worked out.
Sorry about that, but I'll get these issues worked out soon - it shouldn't be too bad to fix

**Comment by Meek97** [comment]
Yeah, the zip archive for `1.1_1.21.1` available from modrinth was just wrong (whoops!)
My bad, but the correct datapack version for 1.21.1 should now be [available here](https://modrinth.com/datapack/better-x-bop/version/1.1_1.21.1). But I did also stumble upon a really weird quirk when using something like [Global Datapacks](https://modrinth.com/mod/datapacks) to automatically inject the datapacks into worlds on world creation.

TL;DR - if you are using something like the Global Datapacks mod, you will need to reboot the world at least once after the initial creation for the BetterNether to properly load the biome customizations.

**Comment by suerion** [comment]
THanks, it is fixed on my side :)
